### PR TITLE
Feature/api work package embedded watchers

### PR DESCRIPTION
**Pull request by mtakac2**
https://www.openproject.org/work_packages/11678

### CODE REVIEW

File: lib/api/v3/users/user_model.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+require 'reform'
+require 'reform/form/coercion'
+
+module API
+  module V3
+    module Users
+      class UserModel < Reform::Form
+        include Coercion
+
+        property :login, type: Integer
```
Comment (tilt) [inline lib/api/v3/users/user_model.rb]: is login an integer?

Comment (mtakac2) [inline lib/api/v3/users/user_model.rb]: Yeah... It is not.. Thanks ;)

File: lib/api/v3/work_packages/work_package_model.rb [hunk 2]
```diff
@@ -104,7 +104,11 @@ def percentage_done=(value)
         end
 
         def activities
-          work_package.journals.map{ |journal| Activities::ActivityModel.new(journal: journal) }
+          work_package.journals.map{ |journal| ::API::V3::Activities::ActivityModel.new(journal: journal) }
+        end
+
+        def users
+          work_package.watcher_users.map{ |u| ::API::V3::Users::UserModel.new(u) }
         end
```
Comment (tilt) [inline lib/api/v3/work_packages/work_package_model.rb:112]: @mtakac wouldn't `watchers` be more appropriate, I know it's rendered as watchers, but why would we name it `users` here?

Comment (mtakac2) [inline lib/api/v3/work_packages/work_package_model.rb:112]: You are right

File: lib/api/v3/users/user_representer.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        include Roar::Representer::JSON::HAL
+        include Roar::Representer::Feature::Hypermedia
+        include Rails.application.routes.url_helpers
+
+        self.as_strategy = API::Utilities::CamelCasingStrategy.new
+
+        property :_type, exec_context: :decorator
+
+        link :self do
+          { href: "#{root_url}/api/v3/users/#{represented.model.id}", title: "#{represented.model.name} - #{represented.model.login}" }
```
Comment (tilt) [inline lib/api/v3/users/user_representer.rb:46]: there's a double `//` rendered after the `root_url` when I let it run locally. Is this due to an incorrect setting oder should we leave the `/` after interpolating here?

File: lib/api/v3/users/user_model.rb (hunk 1 above)
Comment (tilt) [inline lib/api/v3/users/user_model.rb]: @mtakac ok it's not, so we should set it to `string` shouldn't we?

File: lib/api/v3/users/user_representer.rb (hunk 3 above)
Comment (mtakac2) [inline lib/api/v3/users/user_representer.rb:46]: I think it should be set to render the root_url without trailing slash.

Comment (tilt) [inline lib/api/v3/users/user_representer.rb:46]: will you do it before I merge this PR?

Comment (mtakac2) [inline lib/api/v3/users/user_representer.rb:46]: No. That'a not something that is connected to this PR - it's in general settings of OP. I will issue a separate PR.

**Comment by myabc** [comment]
@mtakac shouldn't there be specs for this?

**Pull request merged**